AzureMonitorAgent/agent.py

```diff
@@ -324,6 +324,15 @@ def install():
             if insserv_exit_code != 0:
                 return insserv_exit_code, insserv_output
 
+    # Check if Debian 12 VMs have rsyslog package (required for AMA 1.31+)
+    if (vm_dist.startswith('debian')) and vm_ver.startswith('12'):
+        check_rsyslog, _ = run_command_and_log("dpkg -l | grep rsyslog")
+        if check_rsyslog != 0:
+            hutil_log_info("'rsyslog' package missing from Debian 12 machine, installing to allow AMA to run.")
+            rsyslog_exit_code, rsyslog_output = run_command_and_log("apt update && apt install -y rsyslog")
```

Comment:
in addition to `-y` you may need to set [non-interactive debian frontend](https://askubuntu.com/questions/556385/how-can-i-install-apt-packages-non-interactively)